docs/compression/README.md

```diff
@@ -0,0 +1,76 @@
+# Compression Guide [![Slack](https://slack.minio.io/slack?type=svg)](https://slack.minio.io)
+
+Minio server has compression disabled by default. By enabling it, Minio ensures efficient disk space usage by compressing the objects before being written to disk(s). 
+
+Minio adopts streaming compression format of `golang/snappy`. For better performance, Minio compresses only `compressible objects`. These are objects which will produce a better compression experience. For instance, already compressed objects like `.zip,.bz2` etc, are not fit for compression since they do not have compressible patterns in them.
+
+
+Compression is disabled for
+
+### 1. Non-Compressible objects
+
+    These objects does not produce efficient `Run-length encoding (RLE)` which is a fitness factor for a lossless data compression.
+
+    - Objects having the following extensions
+
+      | `gz` | (GZIP)
+      | `bz2` | (BZIP2)
+      | `rar` | (WinRAR)
+      | `zip` | (ZIP)
+      | `7z` | (7-Zip)
+
+    - Objects having the following content-types
+
+      | `video/*` |
+      | `audio/*` |
+      | `application/zip` |
+      | `application/x-gzip` |
+      | `application/zip` |
+      | `application/x-compress` |
+      | `application/x-spoon` |
+      
+### 2. Encryption enabled requests
+
+    Minio does not support encryption with compression because compression and encryption together enables room for side channel attacks.
```

Comment:
This is not a topic but instead should be mentioned in bold on why it is a bad idea with some citations (links)

Reply:
modified the docs fully